sample/sample-with-di/app/src/main/java/com/dropbox/kaiken/sample/advancedsample/app/UserServices.kt

```diff
@@ -2,50 +2,44 @@ package com.dropbox.kaiken.sample.advancedsample.app
 
 import com.dropbox.kaiken.sample.advancedsample.helloworldfeature.HelloWorldDependencies
 import com.dropbox.kaiken.sample.advancedsample.helloworldfeature.HelloWorldMessageProvider
+import com.dropbox.kaiken.sample.advancedsample.helloworldfeature.HelloWorldMessageProviderUser
 import com.dropbox.kaiken.sample.advancedsample.helloworldfeature.RealWorldMessageProvider
+import com.dropbox.kaiken.sample.advancedsample.helloworldfeature.RealWorldMessageProviderUser
 import com.dropbox.kaiken.sample.advancedsample.helloworldfeature.TimeMessageProvider
+import com.dropbox.kaiken.scoping.AppScope
+import com.dropbox.kaiken.scoping.SingleIn
 import com.dropbox.kaiken.scoping.TeardownHelper
+import com.dropbox.kaiken.scoping.UserScope
 import com.dropbox.kaiken.scoping.UserServices
+import com.squareup.anvil.annotations.ContributesBinding
+import com.squareup.anvil.annotations.ContributesTo
+import com.squareup.anvil.annotations.MergeComponent
 import dagger.BindsInstance
 import dagger.Component
 import dagger.Module
 import dagger.Provides
 
-annotation class UserScope
-
-@Component(
-    modules = [UserServicesModule::class],
-    dependencies = [CoreAppServices::class]
-)
-@UserScope
-abstract class AdvancedKaikenSampleUserServices : UserServices, HelloWorldDependencies {
-    @Component.Factory
-    interface Factory {
-        fun create(
-            appServices: CoreAppServices,
-            @BindsInstance userProfile: UserProfile
-        ): AdvancedKaikenSampleUserServices
-    }
-
-    override fun getTeardownHelper(): TeardownHelper = NOOP_TEARDOWNHELPER
-}
-
 // App Services that are need to instance user services
+@ContributesTo(AppScope::class)
 interface CoreAppServices {
     val timeMessageProvider: TimeMessageProvider
 }
 
+@ContributesTo(UserScope::class)
 @Module
 object UserServicesModule {
+
+
     @Provides
-    @UserScope
-    fun provideHelloWorldMessageProvider(userProfile: UserProfile): HelloWorldMessageProvider =
-        RealWorldMessageProvider("Hello ${userProfile.name}!")
+    @SingleIn(UserScope::class)
+    fun provideHelloWorldMessageProvider(userProfile: UserProfile): HelloWorldMessageProviderUser =
```

Comment:
now that we use subcomponents, we need to qualify which MessageProvider this is, otherwise we conflict with the one that we get passed in from app scope